src/Storage/Field/Type/DateType.php

```diff
@@ -38,7 +51,24 @@ public function query(QueryInterface $query, ClassMetadata $metadata)
     }
 
     /**
-     * @inheritdoc
+     * {@inheritdoc}
+     */
+    public function persist(QuerySet $queries, $entity, EntityManager $em = null)
+    {
+        $key = $this->mapping['fieldname'];
+        $value = $entity->get($key);
+
+        if (!$value instanceof \DateTime && $value !== null) {
+            $value = new Carbon($value);
+            $value::setToStringFormat('Y-m-d');
```

Comment:
Actually you shouldn't even need to do it at all. The format is pulled from the Platform ([here](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Types/DateTimeType.php#L57))